pkg/azuredx/client/client.go

```diff
@@ -141,7 +153,12 @@ func (c *Client) testManagementClient(ctx context.Context, datasourceSettings *m
 		return fmt.Errorf("the Azure cloud '%s' doesn't have the required property for 'resourceManager'", c.cloudSettings.Name)
 	}
 
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, resourceManager+"/providers/Microsoft.ResourceGraph/resources?api-version=2021-03-01", bytes.NewReader(buf))
+	fullUrl, err := url.JoinPath(resourceManager, "/providers/Microsoft.ResourceGraph/resources?api-version=2021-03-01")
```

Comment:
This will have to be what it was before as `JoinPath` doesn't correctly handle the `api-version` query param. Alternatively, the URL could be constructed and the param could be set on the `Query` property